src/unix/linux_like/linux/mod.rs

```diff
@@ -322,7 +322,12 @@ s! {
     }
 
     pub struct input_event {
+        #[cfg(not(gnu_time64_abi))]
         pub time: ::timeval,
+        #[cfg(gnu_time64_abi)]
+        __sec: ::c_ulong,
+        #[cfg(gnu_time64_abi)]
+        __usec: ::c_ulong,
         pub type_: ::__u16,
```

Comment:
This breaks the evdev crate with no obvious way of fixing it (rust-evdev has other errors too, but they are errors I can deal with)

692s error[E0560]: struct `input_event` has no field named `time`
692s    --> src/lib.rs:265:13
692s     |
692s 265 |             time: self.time.as_raw(),
692s     |             ^^^^ `input_event` does not have this field
692s     |
692s     = note: available fields are: `__sec`, `__usec`

The original C header deals with this using macros, which are defined differently for the non-time64 and time64 cases.
(non-time64)

#define input_event_sec time.tv_sec
#define input_event_usec time.tv_usec

(time64)

#define input_event_sec  __sec
#define input_event_usec __usec

Not sure how this can be best dealt with in rust, the best I can think of would be to add get/set/get_mut methods.

Reply:
Thank you for testing.
Does this definition of `input_event` work - it will require changes in using code
```rust
    pub struct input_event {
        #[cfg(any(target_pointer_width = "64", not(gnu_time64_abi)))]
        pub input_event_sec: ::time_t,
        #[cfg(all(target_pointer_width = "32", gnu_time64_abi))]
        pub input_event_sec: ::c_ulong,

        #[cfg(not(gnu_time64_abi))]
        pub input_event_usec: ::suseconds_t,
        #[cfg(gnu_time64_abi)]
        pub input_event_usec: ::c_ulong,

        #[cfg(target_arch = "sparc64")]
        _pad1: u32,

        pub type_: ::__u16,
        pub code: ::__u16,
        pub value: ::__s32,
    }
```

Reply:
Thank you for testing.
Does this definition of `struct input_event` work for evdev-rs - it will require code changes.
```rust
    pub struct input_event {
        #[cfg(any(target_pointer_width = "64", not(gnu_time64_abi)))]
        pub input_event_sec: ::time_t,
        #[cfg(all(target_pointer_width = "32", gnu_time64_abi))]
        pub input_event_sec: ::c_ulong,

        #[cfg(any(target_pointer_width = "64", not(gnu_time64_abi)))]
        pub input_event_usec: ::suseconds_t,
        #[cfg(all(target_pointer_width = "32", gnu_time64_abi))]
        pub input_event_usec: ::c_ulong,

        #[cfg(target_arch = "sparc64")]
        _pad1: ::c_int,

        pub type_: ::__u16,
        pub code: ::__u16,
        pub value: ::__s32,
    }
```

Reply:
Structurally this looks like a sensible compromise and one to which downstream code could be adapted fairly easily.

I still think the definition is probablly wrong on 32-bit linux systems with 64-bit time and C libraries other than glibc.

Reply:
You're probably right, I'll have to think about this some more. 
But I'll update with this struct definition for now.